node/impl/full/eth.go

```diff
@@ -2135,3 +2141,93 @@ func (g gasRewardSorter) Swap(i, j int) {
 func (g gasRewardSorter) Less(i, j int) bool {
 	return g[i].premium.Int.Cmp(g[j].premium.Int) == -1
 }
+
+func (a *EthModule) EthGetBlockReceipts(ctx context.Context, blockParam ethtypes.EthBlockNumberOrHash) ([]*api.EthTxReceipt, error) {
+	ts, err := getTipsetByEthBlockNumberOrHash(ctx, a.Chain, blockParam)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get tipset: %w", err)
+	}
+
+	// Execute the tipset to get the receipts, messages, and events
+	st, msgs, receipts, err := executeTipset(ctx, ts, a.Chain, a.StateAPI)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to execute tipset: %w", err)
+	}
+
+	// Load the state tree
+	stateTree, err := a.StateManager.StateTree(st)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to load state tree: %w", err)
+	}
+
+	// Get the parent tipset CID
+	parentTsCid, err := ts.Parents().Cid()
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get parent tipset cid: %w", err)
+	}
+
+	// Load executed messages and extract events
+	executedMsgs, err := a.EthEventHandler.EventFilterManager.LoadExecutedMessages(ctx, ts, ts)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to load executed messages: %w", err)
+	}
+
+	// Convert executed messages to events
+	events := make([]*filter.CollectedEvent, 0, len(executedMsgs))
+	for _, em := range executedMsgs {
+		for _, ev := range em.Events() {
+			addr, err := address.NewIDAddress(uint64(ev.Emitter))
```

Comment:
We're going to need to check this, does it produce what we want? I suspect not, you'll find out if you compare the output in the itests with `EthGetTransactionReceipt` I think.

there's a translation layer inside the event subsystem to go from id addresses to robust addresses, you can see it in action here: https://github.com/filecoin-project/lotus/blob/9b86cc5d60183251a8d2e8d8999d5d2661e00063/chain/events/filter/event.go#L104-L114

The resolver itself is https://github.com/filecoin-project/lotus/blob/9b86cc5d60183251a8d2e8d8999d5d2661e00063/node/modules/actorevent.go#L134-L147

You could just potentially use the resolver in `EventFilterManager` to do this, it's exposed as `AddressResolver`, just be sure to cache the look-ups like `CollectEvents` does because you're likely to have a lot of duplicates in a single set.